### ISSUE TITLE
How did you build the libraries ?

### ISSUE THREAD

**Issue by hemikak** [Bl0tt0/ApacheQpidDotnetLibraries#1]
Hi @Bl0tt0 , 

How were you able to build the libraries ? Can you give me the commands used in each step. Your help will be appreciated. 

Regards,
@hemika

**Comment by Bl0tt0** [comment]
I basically followed the instructions in the comments of https://issues.apache.org/jira/browse/QPID-6266 modified for VS2013. This is modification is simply replacing the existing options for VS2012 with VS2013. So nothing special. But anyway, here is the documentation I created:

Get the .bat files for compiling both Proton and Qpid here: http://people.apache.org/~chug/proton-win/ 
First download and build Boost.
Please check the scripts in qpid\cpp\bindings\qpid\dotnet if they already contain VS 2013 support. If not, this can simply be added:

configure-windows.ps1
430 } elseif ($vsSelection.Contains("2013")) {
431 $global:vsVersion = "Visual Studio 2013"
432 $global:cmakeGenerator = "Visual Studio 12"
433 $global:vsSubdir = "msvc12"
434 $global:vsSubdirX = "msvcx"
435 $global:cmakeCompiler = "-vc120"
436 $global:vsShortName = "2013"
437 $global:vsEnvironment = """%VS120COMNTOOLS%....\VC\vcvarsall.bat"""

CMakeLists.txt 
190 elseif((${CMAKE_GENERATOR} STREQUAL "Visual Studio 12 2013") OR
191 (${CMAKE_GENERATOR} STREQUAL "Visual Studio 12 Win64"))
192 # Visual Studio 2013
193 select_msvcx_files()
194 set(DOTNET_MSVC_SELECT              "msvcx")
195 set(DOTNET_SLN_FILE_FORMAT          "13.00")
196 set(DOTNET_SLN_VISUAL_STUDIO        "Visual Studio 2013")
197 set(DOTNET_TARGET_FRAMEWORK_VERSION "v4.5")
198 set(DOTNET_PLATFORM_TOOLSET         "<PlatformToolset>v120</PlatformToolset>")

Then execute the .bat scripts. 
**c:\pathToProton\build.bat c:\pathToQpid\qpid 2013 x86**  // note that the qpid checkout has a root directory called qpid. Point to that root directory instead of the checkout directory when building proton
**c:\pathToQpid\build.bat 2013 x86**
Now switch to the build_2013_x86 folder and run start-devenv-messaging-msvc10-x64-64-bit.bat
When building the org.apache.qpid.messaging.sln Visual Studio will fail with an error stating that afxres.h can't be found. Replace the references to afxres.h with WinResrc.h.
After a succesful build you can find the .dll files in build_2013_x86\src\Debug or build_2013_x86\src\Release depending on the selected build configuration

**Comment by Giten2004** [comment]
Did your build result contains "Org.Apache.Qpid.Messaging.SessionReceiver.dll"? I am struggling with the build, some environment issues.

**Comment by Bl0tt0** [comment]
Yes it does. Nevertheless, I will upload an updated version. The current
one only runs on machines that have the windows SDK installed since it is a
debug version. There will be a x86 version in Release config and a x64
version in Release config but without AMQP 1.0 support. Will upload them as
soon as I find the time to update and check the solution

2015-07-13 5:05 GMT+02:00 Michael Xu notifications@github.com:
